### PR TITLE
Update data validation/sanitization

### LINKED ISSUE #11: Data should be sanitized before it's used
All database queries should be rewritten to use `wpdb::prepare()` to make them safer. `$_GET`/`$_POST` values should also be sanitized before they are used.

### PR DESCRIPTION
**Pull request by robincornett**
Fixes #11 

Branched off of issue/12 because of working in the same lines.

Proposed changes:
1. Create `edd_duplicate_product_get_duplicate_url()` helper function to get the URL for duplicating a download.
2. Update row action, edit screen links logic, output.
3. Use `$wpdb->prepare` where possible. I could not figure out how to do this in the `edd_duplicate_post_meta` function.

### CODE REVIEW

**Review by LisaCee — COMMENTED** [review]
Any testing suggestions? Everything looks good and PHPCS is much happier.

Also, do the curly braces added around some query arguments change the query at all or is it more for readability?

**Comment by robincornett** [comment]
I'm not sure I have testing suggestions other than make sure that when you click the link (either on the edit screen or from the download action links, but they should be identical), that the new download looks the same as the old. I should not have changed any functionality, I hope--but post meta, taxonomy terms, things like that should all be copied over seamlessly.

I'm not sure if the `{}` are because of the `""` or because of SQL. But if it is the former, I  know you can glue a variable to a string one of two ways: `'string_' . $variable` or you can do `"string_{$variable}"` which I do personally generally find more readable. I don't know if this is the same scenario, really--but I looked through the EDD core files (especially the newer ones) for similar SQL setup and they use the curly braces.

**Review by LisaCee — COMMENTED** [review]
I'm not 100% sure on the expected behavior for duplication, but the download price copies over to the duplicate in master but not in this branch.

**Comment by robincornett** [comment]
Okay, I think I inadvertently broke the SQL query that I didn't know how to prepare. I think I have it sorted now.

**Review by LisaCee — APPROVED** [review]

File: admin.php [hunk 1]
```diff
@@ -1,43 +1,76 @@
...
+function edd_duplicate_product_link_row( $actions, $post ) {
+	if ( 'download' !== $post->post_type ) {
+		return $actions;
+	}
 
-	$actions['duplicate'] = '<a href="' . wp_nonce_url( admin_url( 'admin.php?action=duplicate_product&amp;post=' . $post->ID ), 'edd-duplicate-product_' . $post->ID ) . '" title="' . __("Make a duplicate from this product", 'edd')
-		. '" rel="permalink">' .  __("Duplicate", 'edd') . '</a>';
+	$actions['duplicate'] = sprintf(
+		'<a href="%s" rel="permalink">%s</a>',
+		edd_duplicate_product_get_duplicate_url( $post->ID ),
```
Comment (ashleyfae) [inline admin.php:17]:
```suggestion
		esc_url( edd_duplicate_product_get_duplicate_url( $post->ID ) ),
```

File: admin.php [hunk 2]
```diff
@@ -1,43 +1,76 @@
...
+ */
+function edd_duplicate_product_get_duplicate_url( $post_id ) {
+
+	$post_id = (int) $post_id;
+
+	return wp_nonce_url(
+		add_query_arg(
+			array(
+				'action' => 'duplicate_product',
+				'post'   => $post_id,
```
Comment (ashleyfae) [inline admin.php:70]:
```suggestion
				'post'   => urlencode( $post_id ),
```

File: duplicate.php [hunk 3]
```diff
@@ -112,19 +140,22 @@ function edd_duplicate_post_taxonomies( $id, $new_id, $post_type ) {
...
-	if (count($post_meta_infos)!=0) {
-		$sql_query = "INSERT INTO $wpdb->postmeta (post_id, meta_key, meta_value) ";
-		foreach ($post_meta_infos as $meta_info) {
-			$meta_key = $meta_info->meta_key;
-			$meta_value = addslashes($meta_info->meta_value);
-			$sql_query_sel[]= "SELECT $new_id, '$meta_key', '$meta_value'";
+	$post_meta_infos = $wpdb->get_results( $wpdb->prepare( "SELECT meta_key, meta_value FROM {$wpdb->postmeta} WHERE post_id = %d", $id ) );
+
+	if ( count( $post_meta_infos ) ) {
+		$sql_query = "INSERT INTO {$wpdb->postmeta} (post_id, meta_key, meta_value) ";
```
Comment (ashleyfae) [inline duplicate.php:152]:
```suggestion
		$sql_query = "INSERT INTO {$wpdb->postmeta} (post_id, meta_key, meta_value) ";
		$sql_query_sel = array();
```

`$sql_query_sel` is never initialized and I'd like it to be.

File: duplicate.php [hunk 4]
```diff
@@ -112,19 +140,22 @@ function edd_duplicate_post_taxonomies( $id, $new_id, $post_type ) {
...
+		foreach ( $post_meta_infos as $meta_info ) {
+			$meta_key        = $meta_info->meta_key;
+			$meta_value      = addslashes( $meta_info->meta_value );
+			$sql_query_sel[] = "SELECT {$new_id}, '{$meta_key}', '{$meta_value}'";
```
Comment (ashleyfae) [inline duplicate.php:156]: If you remove the `addslashes()` on `$meta_value` on line 155, then you should be able to replace this line with:

```suggestion
			$sql_query_sel[] = $wpdb->prepare(
				"SELECT %d, %s, %s",
				$new_id,
				$meta_key,
				$meta_value
			);
```

**Review by ashleyfae — APPROVED** [review]

**Pull request merged**